internal-gateway/envoy.yaml

```diff
@@ -0,0 +1,60 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: internal-gateway-envoy-config
+data:
+  envoy.yaml: |
+    static_resources:
+      listeners:
+      - address:
+          socket_address:
+            address: 0.0.0.0
+            port_value: 80
+        filter_chains:
+        - filters:
+          - name: envoy.filters.network.http_connection_manager
+            typed_config:
+              "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
+              codec_type: AUTO
+              stat_prefix: ingress_http
+              upgrade_configs:
+                - upgrade_type: websocket
+              use_remote_address: true
+              skip_xff_append: false
+              rds:
+                route_config_name: https_routes
+                config_source:
+                  resource_api_version: V3
+                  path_config_source:
+                    path: /config_map/xds/rds.yaml
+                    watched_directory:
+                      path: /config_map/xds
+              http_filters:
+                - name: envoy.filters.http.local_ratelimit
+                  typed_config:
+                    "@type": type.googleapis.com/envoy.extensions.filters.http.local_ratelimit.v3.LocalRateLimit
+                    stat_prefix: http_local_rate_limiter
+                - name: envoy.filters.http.router
+                  typed_config:
+                    "@type": type.googleapis.com/envoy.extensions.filters.http.router.v3.Router
+    admin:
+      address:
+        socket_address:
```

Comment:
Is there a simple example you can point me to so I can work my way up to understanding what's in the yaml?

Reply:
There are lots of examples in the envoy repo and some explanations of terminology in their [documentation](https://www.envoyproxy.io/docs/envoy/v1.23.1/intro/life_of_a_request#terminology), here is one of the smaller examples I could find that shows a [proxy that supports https and sends all traffic to a single http upstream](https://github.com/envoyproxy/envoy/blob/main/examples/tls/envoy-https-http.yaml).

There's a lot of boilerplate but hopefully looking at the `virtual_hosts` section and the `clusters` section give a sense of how things work.